site/src/components/PaginationWidget/PaginationContainer.test.tsx

```diff
@@ -0,0 +1,211 @@
+            isPreviousData: true,
+          }}
+        />,
+      );
+
+      await assertNoScroll(mockScroll);
+    });
+  });
+
+  describe("Responding to changes in React Query's isPreviousData", () => {
+    it("Does nothing when isPreviousData flips from false to true while currentPage stays the same (safety net for 'impossible' case)", async () => {
+      const mockScroll = jest.spyOn(window, "scrollTo");
+
+      const { rerender } = render({
+        paginationUnitLabel: mockUnitLabel,
+        paginationResult: mockInitialRenderResult,
+      });
+
+      rerender(
+        <PaginationContainer
+          paginationUnitLabel={mockUnitLabel}
+          paginationResult={{ ...mockSuccessResult, isPreviousData: true }}
+        />,
+      );
+
+      await assertNoScroll(mockScroll);
+    });
+
+    it("Triggers scroll if scroll has been queued while waiting for isPreviousData to flip from true to false", async () => {
+      const mockScroll = jest.spyOn(window, "scrollTo");
+      const { rerender } = await mountWithSuccess(mockScroll);
+
+      rerender(
+        <PaginationContainer
+          paginationUnitLabel={mockUnitLabel}
+          paginationResult={{
+            ...mockSuccessResult,
+            currentPage: 2,
+            isPreviousData: true,
+          }}
+        />,
+      );
+
+      rerender(
+        <PaginationContainer
+          paginationUnitLabel={mockUnitLabel}
+          paginationResult={{
+            ...mockSuccessResult,
+            currentPage: 2,
+            isPreviousData: false,
+          }}
+        />,
+      );
+
+      await waitFor(() => expect(mockScroll).toBeCalled());
+    });
+
+    it("Cancels a scroll if user interacts with the browser in any way before isPreviousData flips from true to false", async () => {
+      const mockScroll = jest.spyOn(window, "scrollTo");
+
+      // Values are based on (keyof WindowEventMap), but frustratingly, the
+      // native events aren't camel-case, while the fireEvent properties are
+      const userInteractionEvents = [
+        "click",
+        "scroll",
+        "pointerEnter",
+        "touchStart",
+        "keyDown",
+      ] as const;
+
+      for (const event of userInteractionEvents) {
+        const { rerender, unmount } = await mountWithSuccess(mockScroll);
+
+        rerender(
+          <PaginationContainer
+            paginationUnitLabel={mockUnitLabel}
+            paginationResult={{
+              ...mockSuccessResult,
+              currentPage: 2,
+              isPreviousData: true,
+            }}
+          />,
+        );
+
+        fireEvent[event](window);
+
+        rerender(
+          <PaginationContainer
+            paginationUnitLabel={mockUnitLabel}
+            paginationResult={{
+              ...mockSuccessResult,
+              currentPage: 2,
+              isPreviousData: false,
+            }}
+          />,
+        );
+
+        await assertNoScroll(mockScroll);
+        unmount();
```

Comment:
Why do we have to explicitly unmount?

Reply:
Because the loop creates a new version of the UI per iteration. But if you don't unmount the previous one, the old UI from the last render sticks around and can cause some issues with test utilities

This happens automatically for most test cases, but it needs to be done manually here because multiple renders are happening before the test case can get garbage-collected